src/platform/backends/qemu/linux/qemu_platform_detail_linux.cpp

```diff
@@ -187,6 +188,15 @@ QStringList mp::QemuPlatformDetail::vm_platform_args(const VirtualMachineDescrip
                                             tap_device_name, vm_desc.default_mac_address));
 }
 
+// FIXME: after moving to core22, this will be handled by dnsmasq --dhcp-ignore-clid
+void multipass::QemuPlatformDetail::release_mac_with_different_hostname(const std::string& hw_addr,
+                                                                        const std::string& name)
+{
+    if (auto ip_and_host = dnsmasq_server->get_ip_and_host_for(hw_addr);
+        ip_and_host.has_value() && ip_and_host.value().second != name)
+        dnsmasq_server->release_mac(hw_addr);
```

Comment:
I know this is very picky, but I couldn't understand this at first sight because of the unfortunate indentation. What about:
```
    if (auto ip_and_host = dnsmasq_server->get_ip_and_host_for(hw_addr); ip_and_host && ip_and_host->second != name)
        dnsmasq_server->release_mac(hw_addr);
```
?

The whole `if` block would fit in one line. It is completely valid to test the `std::optional` for nullity and the `operator*` is equivalent to `value()` when the optional contains a value.

Reply:
Indeed it is not very readable. Thanks for the tips!